src/training/save_states.rs

```diff
@@ -30,163 +30,167 @@ static mut SAVE_STATE_SITUATION_KIND_CPU: i32 = 0 as i32;
+
+        // force aerial, because from aerial state we can move anywhere
+        if StatusModule::situation_kind(module_accessor) == SITUATION_KIND_GROUND {
+            StatusModule::change_status_request(module_accessor, *FIGHTER_STATUS_KIND_JUMP_SQUAT, false);
         }
+        return;
+    }
 
-        // move to camera bounds
-        if *save_state == CameraMove {
-            *save_state = PosMove;
-
-            let left_right =
-                (PostureModule::pos_x(module_accessor) > 0.0) as i32 as f32 - (PostureModule::pos_x(module_accessor) < 0.0) as i32 as f32;
-            let y_pos = 20.0;
-
-            let pos = Vector3f {
-                x: left_right * 50.0,
-                y: y_pos,
-                z: 0.0,
-            };
-            PostureModule::set_pos(module_accessor, &pos);
-
-            // force aerial, because from aerial state we can move anywhere
-            if StatusModule::situation_kind(module_accessor) == SITUATION_KIND_GROUND {
-                StatusModule::change_status_request(module_accessor, *FIGHTER_STATUS_KIND_JUMP_SQUAT, false);
-            }
+    // move to correct pos
+    if *save_state == PosMove {
+        if StatusModule::situation_kind(module_accessor) == SITUATION_KIND_GROUND {
             return;
         }
 
-        // move to correct pos
-        if *save_state == PosMove {
-            if StatusModule::situation_kind(module_accessor) == SITUATION_KIND_GROUND {
-                return;
-            }
-
-            KineticModule::clear_speed_all(module_accessor);
-
-            let pos = Vector3f {
-                x: *save_state_x,
-                y: *save_state_y,
-                z: 0.0,
-            };
-            PostureModule::set_pos(module_accessor, &pos);
-            PostureModule::set_lr(module_accessor, *save_state_lr);
-            DamageModule::heal(
+        KineticModule::clear_speed_all(module_accessor);
+
+        let pos = Vector3f {
+            x: *save_state_x,
+            y: *save_state_y,
+            z: 0.0,
+        };
+        PostureModule::set_pos(module_accessor, &pos);
+        PostureModule::set_lr(module_accessor, *save_state_lr);
+        DamageModule::heal(
+            module_accessor,
+            -1.0 * DamageModule::damage(module_accessor, 0),
+            0,
+        );
+        DamageModule::add_damage(module_accessor, *save_state_percent, 0);
+
+        StatusModule::set_situation_kind(
+            module_accessor,
+            app::SituationKind {
+                situation_kind: *save_state_situation_kind,
+            },
+            false,
+        );
+
+        if *save_state_situation_kind == SITUATION_KIND_GROUND
+            && status != FIGHTER_STATUS_KIND_WAIT
+        {
+            StatusModule::change_status_request(
                 module_accessor,
-                -1.0 * DamageModule::damage(module_accessor, 0),
-                0,
+                *FIGHTER_STATUS_KIND_CLIFF_WAIT,
```

Comment:
This is wrong, should be `FIGHTER_STATUS_KIND_WAIT`. This is for if you're resetting either player to the idle state.